File: tests/php/testcases/NeedsAuthTest.php
<?
require_once 'base.php';

class NeedsAuthTestCase extends TestServiceTestCase {

    public function test_needs_context() {
        $headers = new Headers();
        $headers->username = 'user';
        $headers->password = 'pass';
        $this->service->__setSoapHeaders(
            new SoapHeader('http://example.com/', 'headers', $headers));
        $this->assertEquals(
            $this->service->say_hello(), 'hello user');
    }

    public function test_needs_context_exception() {
        $headers = new Headers();
        $headers->username = 'fake';
        $headers->password = 'pass';
        $this->service->__setSoapHeaders(
            new SoapHeader('http://example.com/', 'headers', $headers));
        try {
            $this->service->say_hello();
        } catch (SoapFault $e) {
            $this->assertEquals(
                $e->getMessage(), 'Access denied for user fake');
            return;
        }
        $this->assertTrue(false, 'Exception hasn\'t been thrown');
    }
}

File: tests/mono/testcases/NeedsAuthTest.cs
using System;
using NUnit.Framework;

namespace TestCases {

    [TestFixture()]
    public class NeedsAuthTest : TestServiceTest {

        [Test()]
        public void test() {
            Headers headers = new Headers();
            headers.username = "user";
            headers.password = "pass";
            service.HeadersValue = headers;
            Assert.AreEqual("hello user", service.say_hello());
        }

        [Test()]
        public void test_exception() {
            Headers headers = new Headers();
            headers.username = "fake";
            headers.password = "pass";
            service.HeadersValue = headers;
            try {
                service.say_hello();
            } catch(Exception e) {
                Assert.AreEqual("Access denied for user fake", e.Message);
                return;
            }
            Assert.True(false, "Exception hasn't been thrown");
        }
    }
}
